app/gui/view/graph-editor/src/component/node.rs

```diff
@@ -323,7 +323,7 @@ ensogl::define_endpoints_2! {
         /// Press event. Emitted when user clicks on non-active part of the node, like its
         /// background. In edit mode, the whole node area is considered non-active.
         background_press         (),
-        expression               (enso_text::Rope),
+        expression               (String),
```

Comment:
Why this change? Passing String in FRP is ultra slow. It requires cloning the string on each FRP node. If you really need to pass String, use ImString instead.

Reply:
I changed it to `ImString`. I can't use `Rope` because, in some code paths, it is not available now.

Reply:
What od you mean by "not available now"?

Reply:
So before code changes `expression` was only produced by `Text::content` FRP output. Now we emit `expression` two times: when the user changes the code (Text::content) and when the code is changed by the Presenter (`set_expression` input). The code behavior is roughly the same, but we don't have `Rope` in the second case, we only have `ImString` from `Expression::code`.